src/mix_protocol.nim

```diff
@@ -49,123 +52,117 @@ proc cryptoRandomInt(max: int): Result[int, string] =
 
-      let parts = fullAddrStr.split("/p2p/")
-      if parts.len != 2:
-        error "Invalid multiaddress format", parts = parts
-        return
+    let
+      locationAddrStr = parts[0]
+      peerIdStr = parts[1]
 
-      let
-        locationAddrStr = parts[0]
-        peerIdStr = parts[1]
+    # Create MultiAddress and PeerId
+    let locationAddr = MultiAddress.init(locationAddrStr).valueOr:
+      error "Failed to parse location multiaddress: ", err = error
+      return
 
-      # Create MultiAddress and PeerId
-      let locationAddr = MultiAddress.init(locationAddrStr).valueOr:
-        error "Failed to parse location multiaddress: ", err = error
-        return
+    let peerId = PeerId.init(peerIdStr).valueOr:
+      error "Failed to initialize PeerId", err = error
+      return
 
-      let peerId = PeerId.init(peerIdStr).valueOr:
-        error "Failed to initialize PeerId", err = error
-        return
-
-      var destConn: Connection
-      try:
-        destConn = await mixProto.switch.dial(peerId, @[locationAddr], $protocol)
-        await destConn.writeLp(message)
-      except CatchableError as e:
-        error "Failed to dial next hop: ", err = e.msg
-    of Success:
-      info "# Intermediate: ", multiAddr = multiAddr
-      # Add delay
-      let delayMillis = (delay[0].int shl 8) or delay[1].int
-      await sleepAsync(milliseconds(delayMillis))
-
-      # Forward to next hop
-      let nextHopBytes = getHop(nextHop)
-
-      let fullAddrStr = bytesToMultiAddr(nextHopBytes).valueOr:
-        error "Failed to convert bytes to multiaddress", err = error
-        return
-
-      let parts = fullAddrStr.split("/p2p/")
-      if parts.len != 2:
-        error "Invalid multiaddress format", parts = parts
-        return
-
-      let
-        locationAddrStr = parts[0]
-        peerIdStr = parts[1]
-
-      # Create MultiAddress and PeerId
-      let locationAddr = MultiAddress.init(locationAddrStr).valueOr:
-        error "Failed to parse location multiaddress: ", err = error
-        return
-
-      let peerId = PeerId.init(peerIdStr).valueOr:
-        error "Failed to initialize PeerId", err = error
-        return
-
-      var nextHopConn: Connection
-      try:
-        nextHopConn = await mixProto.switch.dial(peerId, @[locationAddr], MixProtocolID)
-        await nextHopConn.writeLp(processedPkt)
-      except CatchableError as e:
-        error "Failed to dial next hop: ", err = e.msg
-    of Duplicate:
-      discard
-    of InvalidMAC:
-      discard
+    var nextHopConn: Connection
+    try:
+      nextHopConn = await mixProto.switch.dial(peerId, @[locationAddr], MixProtocolID)
+      await nextHopConn.writeLp(processedPkt)
+    except CatchableError as e:
+      error "Failed to dial next hop: ", err = e.msg
+    finally:
+      if nextHopConn != nil:
+        try:
+          await nextHopConn.close()
+        except CatchableError as e:
+          error "Failed to close outgoing stream: ", err = e.msg
```

Comment:
This had a _really_ weird conflict to resolve. Need confirmation that the resulting code here is sane.